src/diagnostics.md

```diff
@@ -331,15 +331,6 @@ To solve this, `libsyntax` defines its own buffered lint type, which
 `ParseSess::buffer_lint` uses. After macro expansion, these buffered lints are
 then dumped into the `Session::buffered_lints` used by the rest of the compiler.
 
-Usage for buffered lints in `libsyntax` is pretty much the same as the rest of
-the compiler with one exception because we cannot import the `LintId`s for
-lints we want to emit. Instead, the [`BufferedEarlyLintId`] type is used. If you
-are defining a new lint, you will want to add an entry to this enum. Then, add
-an appropriate mapping to the body of [`Lint::from_parser_lint_id`][fplid].
-
-[`BufferedEarlyLintId`]: https://doc.rust-lang.org/nightly/nightly-rustc/syntax/early_buffered_lints/enum.BufferedEarlyLintId.html
-[fplid]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc/lint/struct.Lint.html#method.from_parser_lint_id
```

Comment:
This section seems outdated since https://github.com/rust-lang/rust/commit/f03d8f305a80778ae034b0205803ea3edc297ac8. Could someone create a follow-up PR?

Reply:
I believe @Mark-Simulacrum said they would do that.

Reply:
Yes, I will, just haven't gotten around to it yet this morning. Hopefully soon!

Reply:
Turns out these updates are pretty much perfect as the changes made in the relevant rustc PR made early lints much more like late lints in the sense that there's nothing special about how to register them.